invenio_rdm_records/services/generators.py

```diff
@@ -155,6 +156,26 @@ def _condition(self, record, file_key=None, **kwargs):
         return is_file_local
 
 
+# class IfRecordWithExternalDOI(ConditionalGenerator):
+#     """Generator that depends on whether the record has external DOI or not."""
+#
+#     def _condition(self, record, **kwargs):
+#         """Check if the record has external DOI."""
+#         is_feature_enabled = not current_app.config["RDM_LOCK_FILES_FOR_EXTERNAL_DOI"]
```

Comment:
I'm also wondering if RDM_UNLOCK_FILES_EXTERNAL_DOI might be a bit clearer, since we could avoid the `not` and map it clearly to the feature. Then the other one would be RDM_DISABLE_VERSIONS_EXTERNAL_DOI